src/libcommon/utility/types.h

```diff
@@ -211,14 +230,30 @@ enum class ExitCause {
     LiteSyncNotAllowed,
     NetworkTimeout,
     SocketsDefuncted, // macOS: sockets defuncted by kernel
-    NoSearchPermission,
     NotFound,
     QuotaExceeded,
     FullListParsingError,
     OperationCanceled
 };
 std::string toString(ExitCause e);
 
+struct ExitInfo {
+        ExitInfo() = default;
+        constexpr ExitInfo(const ExitCode &code, const ExitCause &cause) : _code(code), _cause(cause) {}
+        ExitInfo(const ExitCode &code) : _code(code) {}
+        const ExitCode &code() const { return _code; }
+        const ExitCause &cause() const { return _cause; }
+        operator ExitCode() const { return _code; }
+        operator ExitCause() const { return _cause; }
+        constexpr operator bool() const { return _code == ExitCode::Ok; }
+        constexpr explicit operator int() const { return toInt(_code) * 100 + toInt(_cause); }
```

Comment:
That is a nice idea. But should we set the `Ok` value to 0 then so any value higher than 0 is an anomaly?

Reply:
Done here: [1aa5ba2](https://github.com/Infomaniak/desktop-kDrive/pull/356/commits/1aa5ba28e6a2a7b941c49ca8e95727b906eb9551)
We have already checked with Christophe. The only place in the database where we store the exit code/cause is in the Error table, which is cleared with each sync restart.